tethysext/atcore/models/file_database/file_database_client.py

```diff
@@ -0,0 +1,83 @@
+"""
+********************************************************************************
+* Name: file_database_client.py
+* Author: glarsen
+* Created On: November 10, 2020
+* Copyright: (c) Aquaveo 2020
+********************************************************************************
+"""
+import os
+import logging
+import uuid
+
+from sqlalchemy.orm.session import Session
+
+from tethysext.atcore.mixins.meta_mixin import MetaMixin
+from tethysext.atcore.models.file_database import FileDatabase
+
+log = logging.getLogger('tethys.' + __name__)
+
+
+class FileDatabaseClient(MetaMixin):
+    def __init__(self, session: Session, file_database_id: uuid.UUID):
+        """Init function for the FileDatabaseClient"""
+        self._database_id = file_database_id
+        self._instance = None
+        self._session = session
+        self._path = None
+
+    @classmethod
+    def new(cls, session: Session, root_directory: str, meta: dict = None) -> 'FileDatabaseClient':
+        """
+        Class method for creating a new instance of the FileCollectionClient class.
+
+        Args:
+            session: The session for the database.
+            root_directory (str): Directory for the files for this FileDatabase
+            meta (dict): The meta for the FileCollection
+        """
+        meta = meta or {}
+        new_file_database = FileDatabase(
+            root_directory=root_directory,
+            meta=meta,
+        )
+        session.add(new_file_database)
+        session.commit()
+        client = cls(session, new_file_database.id)
+
+        client.write_meta()
+
+        return client
+
+    @property
+    def instance(self) -> FileDatabase:
+        """Property to get the underlying instance so it can be lazy loaded."""
+        if not self._instance:
+            self._instance = self._session.query(FileDatabase).get(self._database_id)
+        return self._instance
+
+    @property
+    def meta(self) -> dict:
+        """Property to get the meta from the underlying instance."""
+        return self.instance.meta
```

Comment:
I'm thinking about this case:

```python
client.meta['foo'] = 'bar'
```
They get a handle on the meta object via the getter, and change it in place. This is a change to meta, but we wouldn't be able to sync to the db or meta.json file .

Reply:
We can disallow it by returning a deepcopy of `self.instance.meta`. Don't know that there is another way to handle it. That would mean users would need to modify meta like this:

```python
my_meta = client.meta
my_meta['foo'] = 'bar'
client.meta = my_meta
```
Which is sort of annoying. Can we override the `[ ]` operator or something like that?

Reply:
I guess to do ^ that we'd need to return some wrapper object around meta. Gah. IDK. What do you think?